src/WebUI/Common/KubeSummary.tsx

```diff
@@ -218,6 +251,16 @@ export class KubeSummary extends BaseComponent<IKubeSummaryProps, IKubernetesCon
         this._selectedItemViewMap[SelectedItemKeys.ReplicaSetKey] = (item) => this._getWorkoadPodsViewComponent(item.metadata, item.spec && item.spec.template, item.kind || "ReplicaSet", item.status.availableReplicas, item.status.replicas);
     }
 
+    private _populateObjectFinder() {
+        this._objectFinder[SelectedItemKeys.ReplicaSetKey] = (uid) => (this._workloadsStore.getState().replicaSetList as V1ReplicaSetList).items.filter(r => r.metadata.uid == uid)[0];
```

Comment:
as mentioned in m148 this can be optimized.
also should we have lowercase, so we do not have a problem with casing?

Reply:
i'll sync up with you.